File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Ram/AlibabaCloud/SDK/ROS/CDK/Ram/RosGroupProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Ram
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::RAM::Group`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-ram.RosGroupProps")]
    public class RosGroupProps : AlibabaCloud.SDK.ROS.CDK.Ram.IRosGroupProps
    {
        /// <remarks>
        /// <strong>Property</strong>: groupName: Specifies the group name, containing up to 64 characters.
        /// </remarks>
        [JsiiProperty(name: "groupName", typeJson: "{\"primitive\":\"string\"}", isOverride: true)]
        public string GroupName
        {
            get;
            set;
        }

        /// <remarks>
        /// <strong>Property</strong>: comments: Remark information, up to 128 characters or Chinese characters.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "comments", typeJson: "{\"primitive\":\"string\"}", isOptional: true, isOverride: true)]
        public string? Comments
        {
            get;
            set;
        }

        /// <remarks>
        /// <strong>Property</strong>: policies: Describes what actions are allowed on what resources.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "policies", typeJson: "{\"union\":{\"types\":[{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"},{\"collection\":{\"elementtype\":{\"union\":{\"types\":[{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"},{\"fqn\":\"@alicloud/ros-cdk-ram.RosGroup.PoliciesProperty\"}]}},\"kind\":\"array\"}}]}}", isOptional: true, isOverride: true)]
        public object? Policies
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdkram/RosGroupProps.go
package alicloudroscdkram


// Properties for defining a `RosGroup`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-ram-group
type RosGroupProps struct {
	GroupName interface{} `field:"required" json:"groupName" yaml:"groupName"`
	Comments interface{} `field:"optional" json:"comments" yaml:"comments"`
	DeletionForce interface{} `field:"optional" json:"deletionForce" yaml:"deletionForce"`
	IgnoreExisting interface{} `field:"optional" json:"ignoreExisting" yaml:"ignoreExisting"`
	Policies interface{} `field:"optional" json:"policies" yaml:"policies"`
	PolicyAttachments interface{} `field:"optional" json:"policyAttachments" yaml:"policyAttachments"`
}

